Packs/JoeSecurity/Integrations/JoeSecurityV2/JoeSecurityV2.py

```diff
@@ -0,0 +1,527 @@
+
+         Args:
+            args: Dict(str, any): pagination arguments (page, page_size, limit).
+            results: (Generator): API results for pagination.
+
+         Returns:
+             result: List: The requests pages.
+    """
+    page = arg_to_number(args.get('page', None))
+    page_size = arg_to_number(args.get('page_size', None))
+    limit = arg_to_number(args.get('limit', 50))
+
+    # pagination is available only if supplied page and page size.
+    if (page and not page_size) or (page_size and not page):
+        raise Exception("one of the page or page_size arguments are missing")
+    if (page and page <= 0) or (page_size and page_size <= 0) or (limit and limit < 0):
+        raise Exception("one of the arguments are not having a valid value")
+
+    number_of_entries = page * page_size if (page and page_size) else limit if limit else 0
+    try:
+        all_pages = [next(results) for _ in range(0, number_of_entries)]
+    except Exception:
+        return []
+    return all_pages[(-1) * int(page_size):] if page_size else all_pages
+
+
+def build_analysis_hr(analysis: Dict[str, Any]) -> Dict[str, Any]:
+    """
+         Helper function that supports the building of the human-readable output.
+
+         Args:
+            analysis: Dict(str, any): Analysis result returned by the API.
+
+         Returns:
+             result: Dict(str, any): The analysis human-readable entry.
+    """
+    file_name = analysis.get('filename')
+    sha1 = analysis.get('sha1')
+    sha256 = analysis.get('sha256')
+    md5 = analysis.get('md5')
+    tags = analysis.get('tags')
+    hr_analysis = {
+        'ID': analysis.get('webid'),
+        'SampleName': file_name,
+        'Status': analysis.get('status'),
+        'Time': analysis.get('time'),
+        'MD5': md5,
+        'SHA1': sha1,
+        'SHA256': sha256,
+        'Systems': list(set([run.get('system') for run in analysis.get('runs', [])])),
+        'Result': list(set([run.get('detection') for run in analysis.get('runs', [])])),
+        'Tags': tags,
+        'Errors': list(set([run.get('error') for run in analysis.get('runs', [])])),
+        'Comments': analysis.get('comments'),
+    }
+    return hr_analysis
+
+
+def build_reputation_hr(analysis: Dict[str, Any], command: str) -> Dict[str, Any]:
+    """
+          Helper function that supports the building of the human-readable output.
+
+          Args:
+             analysis: (Dict(str, any)): Analysis result returned by the API.
+             command: (str): The command url or file.
+
+          Returns:
+              result: Dict(str, any): The analysis human-readable entry.
+     """
+    if command == 'file':
+        file_name = analysis.get('filename')
+        sha1 = analysis.get('sha1')
+        sha256 = analysis.get('sha256')
+        md5 = analysis.get('md5')
+        tags = analysis.get('tags')
+        hr_analysis = {
+            'MeaningfulName': file_name,
```

Comment:
let's make sure this complies with our [context standards](https://xsoar.pan.dev/docs/integrations/context-standards-mandatory)

Reply:
It is the human-readable so I don't think we have a standard for it but I will change it to `File Name`